Doc/library/sqlite3.rst

```diff
@@ -2358,6 +2345,88 @@ can be found in the `SQLite URI documentation`_.
 .. _SQLite URI documentation: https://www.sqlite.org/uri.html
 
 
+.. _sqlite3-howto-row-factory:
+
+How to create and use row factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` represents each fetched row as a :class:`tuple`.
+If a :class:`!tuple` does not suit your needs,
+use the :class:`sqlite3.Row` class or a custom :attr:`~Cursor.row_factory`.
+
+Even though :attr:`!row_factory` exists as an attribute both on the
+:class:`Cursor` and the :class:`Connection`,
+it is recommended to set it as :class:`Connection.row_factory`,
+so all cursors created off of the connection will use the same row factory.
+
+:class:`!Row` provides indexed and case-insensitive named access to columns,
+with low memory overhead and minimal performance impact.
+In order to use :class:`!Row` as a row factory,
+simply assign it to the :attr:`Connection.row_factory` attribute:
+
+.. doctest::
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = sqlite3.Row
+
+Query results are now returned as :class:`!Row` instances:
+
+.. doctest::
+
+   >>> res = con.execute("SELECT 'Earth' AS name, 6378 AS radius")
+   >>> row = res.fetchone()
+   >>> row.keys()
+   ['name', 'radius']
+   >>> row[0]  # Access by index.
```

Comment:
```suggestion
   >>> row[0]         # Access by index.
```

If you're going to line the others up rather than using two spaces, seems like you should line this one up too, especially since the close parallel is between this one and the next.